flang/lib/Lower/RTBuilder.h

```diff
@@ -209,6 +211,24 @@ constexpr TypeBuilderFunc getModel<const Fortran::runtime::Descriptor &>() {
   };
 }
 template <>
+constexpr TypeBuilderFunc getModel<Fortran::runtime::Descriptor &>() {
+  return [](mlir::MLIRContext *context) -> mlir::Type {
+    return fir::ReferenceType::get(
+        fir::BoxType::get(mlir::NoneType::get(context)));
+  };
+}
+template <>
+constexpr TypeBuilderFunc getModel<Fortran::runtime::Descriptor *>() {
+  return getModel<Fortran::runtime::Descriptor &>();
+}
+template <>
+constexpr TypeBuilderFunc getModel<Fortran::common::TypeCategory>() {
+  return [](mlir::MLIRContext *context) -> mlir::Type {
+    // FIXME: use enum underlying type.
```

Comment:
can this be fixed now? We should be able to use `sizeof(enum) * 8` or something here -- though a comment that the code assumes `build == host == target` would be warranted.

Reply:
Sure, I added a TODO note on the top of the file because they are a number of other helpers that are using sizeof.